doc/content/applications/se-vanilla/uplinks.md

```diff
@@ -0,0 +1,45 @@
+---
+title: "uplinks"
+description: ""
+weight: 1
+---
+
+The vanilla app uplinks are compromised of two simple messages (heartbeat and temperature/humidity) to reduce the complexity and show a simple example of the device functionality.
+
+## Payload formatters
+
+Please see [The Things Stack Javascript payload formatter documentation](https://www.thethingsindustries.com/docs/integrations/payload-formatters/javascript/) for information on how to use payload formatters.
+
+The javascript payload formatter below can be used to decode the vanilla app.
+
+```javascript
+function decodeUplink(input) {
```

Comment:
Can be formatted to:
```javascript
function decodeUplink(input) {
    var data = {};
    if (input.fPort == 1) {
        data.batteryVoltage = (input.bytes[0] / 10);
        data.fwVersion = input.bytes[1] + (input.bytes[2] / 10);
    }
    if (input.fPort == 2) {
        data.temperature = ((input.bytes[0] << 8) + input.bytes[1]) / 10;
        data.humidity = ((input.bytes[2] << 8) + input.bytes[3]) / 10;
    }
    return {
        data: data,
    };
}
```

Maybe worth adding to the Device Repository instead ? For the old Uno/Nodes, we allow people to select one of the default sketches as a firmware version, so maybe we could have the production app as one of the available firmwares, and it could use this formatter.

Reply:
With the current design of our GNSE repo, I don't recommend having a production app as standalone app.

The reason for this is that the applications were designed to allow for flexibility and versatility.
Adding a production app there will actively diminish the other applications purpose.

With sensors_lorawan and secure_element_lorawan I think users have all the foundation they need to build something similar to the production app.

I would like to also highlight that this this device addresses a different segment of developers than the ones using UNOs and The Things Node.

Reply:
> Maybe worth adding to the Device Repository instead ?

I think it should be in both